lms/djangoapps/student_account/views.py

```diff
@@ -487,6 +487,8 @@ def account_settings_context(request):
             # If the user is connected, sending a POST request to this url removes the connection
             # information for this provider from their edX account.
             'disconnect_url': pipeline.get_disconnect_url(state.provider.provider_id, state.association_id),
-        } for state in auth_states]
+            # We only want to include providers if they are either currently available to be logged
+            # in with, or if the user is already authenticated with them.
+        } for state in auth_states if state.provider.display_for_login or state.has_account]
```

Comment:
Not really. If the student's account with the IdP is invalid, then we'll never get the postback from the IdP during setup. If the student's account with the IdP becomes invalid later on, then it doesn't really matter - we've already got the identification information, so the worst-case scenario is that the student becomes unable to log in through the IdP, because it doesn't send us a signin postback. If that's the case, I _believe_, but am not 100% sure, that the student can still reset their password using the email they provided at registration and sign in the boring way.